applications/nrf_desktop/README.rst

```diff
@@ -32,16 +26,64 @@ This means that parts of application functionality were separated into isolated
 Application events are handled by the :ref:`event_manager`.
 Modules register themselves as listeners of events they need to react on.
 
+The following figure shows the nRF Desktop modules and their relation with other components and the :ref:`event_manager`.
+
+.. figure:: /images/nrf_desktop_arch.svg
+   :scale: 25 %
+   :alt: nRF Desktop HLD
+
+   Application high-level design overview (click to enlarge)
+
 The application limits the number of threads used to the minimum and does not use user-space threads.
-Most of the application activity happens in the context of the system workqueue thread via scheduled work objects or the event manager callbacks (executed from the system workqueue thread).
+Most of the application activity happens in the context of the system workqueue thread either through scheduled work objects or through the event manager callbacks (executed from the system workqueue thread).
 
-Firmware modules:
+Since the application architecture is uniform and the code is shared, the set of modules in use depends on the selected device role.
+A different set of modules is enabled when the application is working as a mouse, a keyboard, or as a dongle.
 
-.. toctree::
-    :maxdepth: 1
+Modules used for gaming mouse
```

Comment:
```suggestion
Gaming mouse modules
```
I think this looks much better as a section header. If you agree, please change for the next two sections as well and also adjust the length of "--------" under these headers.